add notes for n26441 (#623)
bitcoin-core-review-club/website: _posts/2023-02-15-#26441.md

commit:
---

## Notes

- A Bitcoin peer may have connections with other peers. By default (in Bitcoin Core), it allows up to 125 connections, 
  8 of which are outbound.

- There are different kind of connections. E.g: block-relay-only, inbound, outbound-full-relay, manual, feeler, 
  block-relay and addr-fetch.

- You can use RPC `getpeerinfo` to get data about each connected network peer and check the connection type, permissions
  and other informations.

> - `-whitelist` flag allows to add permission flags to the peers connecting from the given IP address or CIDR-notated

Reviewer: whitelisting and permissions are sometimes used interchangeably. Perhaps it's useful to expand a bit on this in the notes?

Author: you mean `-whitelist` and `-whitebind`?

  network. It uses the same permissions as `-whitebind`: bloomfilter, noban, forcerelay, relay, mempool, download,
  addr).

> - #17167 proposed to change `-whitelist` to allow whitelisting outgoing connections.

Author: The approach hasn't been changed, just grabbed #17167 and added the RPC command.

## Questions

1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)? 
   What was your review approach?

1. What does this PR do?

1. `-whitelist` only allows to add permission flags to inbound peers. Why only for inbound ones?
   

> 1. Does it make sense to extend the permissions to outbound peers? Why?

Reviewer: Isn't this the same as the previous question?

Author: Not the same, but I think we can merge them.

> 1. Considering we already have `-whitelist`, why would an RPC be useful? What we want to avoid?

Reviewer: ```suggestion
1. Considering we already have the `-whitelist` startup option, why would an RPC be useful? What do we want to avoid?
```

> 1. This PR adds a `ConnectionDirection` parameter in `TryParsePermissionFlags` to control whether it will apply the permissions to inbound or 

Reviewer: I added a [tag](https://github.com/bitcoin-core-review-club/bitcoin/releases/tag/pr26441) for the current head to bitcoin-core-review-club/bitcoin so these links are persisted:
```suggestion
1. This PR [adds](https://github.com/bitcoin-core-review-club/bitcoin/commit/7dc3ac9cd5fcbb1da42f57fd02aa2f3dff325dd3#diff-bcb9f9354b59da783335c9a2b9d706fd5679e94bf63d67a17959544ef7255343R23-R24) a `ConnectionDirection` parameter in `TryParsePermissionFlags` to control whether it will apply the permissions to inbound or 
```

You can update line 10 to set the `commit` to 153e86e

>    outbound connections. In [netbase.h](https://github.com/bitcoin/bitcoin/blob/master/src/netbase.h#L38), `ConnectionDirection` has 2 

Reviewer: Best to use permalinks, otherwise it goes out of date pretty quickly

   operators overloading. Could you explain how Operator Overloading in C++ works and how it has been used in `ConnectionDirection`?

> 1. `ConnectionDirection` can be `Both`, `In`, `Out` or `None`. What happens in `TryParsePermissionFlags` if its `None`? What scenarios 
>    it can happen?

Reviewer: ```suggestion
1. `ConnectionDirection` can be `Both`, `In`, `Out` or `None`. What happens in `TryParsePermissionFlags` if it is `None`? In which scenarios 
   can this happen?
```

> 1. In `addpermissionflags` RPC we receive an array of permission flags and the ip (or network). However, we convert it to a string with the following

Reviewer: ```suggestion
1. In the `addpermissionflags` RPC we receive an array of permission flags and the IP (or network). However, we convert it to a string of the following
```
